Add nsoftware PKI Proxy instructions for NetHSM

mmerklinger opened the pull request:
This PR adds the setup instructions for /n software PKI Proxy with NetHSM.

mmerklinger committed:
Add nsoftware PKI Proxy instructions for NetHSM

jans23 commented:
I suggest to add clear recommendation to use our PKCS#11 (without PKI Proxy) in normal scenarios and in which scenario PKI Proxy's PKCS#11 interface may be useful.

mmerklinger commented:
I added a two examples for PKI Proxy PKCS#11 usage.

jans23 commented on source/components/nethsm/pkiproxy.rst:
```diff
@@ -0,0 +1,137 @@
+KSP (Key Storage Provider)
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+PKI Proxy provides a KSP to interface with the PKI Proxy server from a client, through native Windows APIs.
```
Unsure what the term "client" refers to exactly. It's not a PKI Proxy Client but an arbitrary application, right?

mmerklinger replied:
It was meant in the context of the "PKI Proxy - Client" chapter as a machine (client) which runs the PKI Proxy client tools. I shortened it, as the client tools part should be understandable from the context of the chapter.

jans23 commented on source/components/nethsm/pkiproxy.rst:
```diff
@@ -0,0 +1,138 @@
+PKI Proxy provides a PKCS#11 module to interface with the PKI Proxy server.
+Please refer to the `PKI Proxy documentation <https://cdn.nsoftware.com/help/PK1/app/nsoftware.PKIProxy.htm#pg_runningpkcs11>`__ for more information.
+
+.. important::
```
I suggest to move this information to the beginning, incorporate it into line 6 and subsequent. This way the reader can understand at the beginning (and not at the end) if these instructions are applicable to him/her.

jans23 commented:
So that this page can be found easier we should use the term "CNG" where suitable.

jans23 commented:
And maybe we should also write the long terms: Cryptography Next Generation (CNG), Key Storage Provider (KSP)

mmerklinger commented:
I included a sentence about the CNG in the introduction and also mentioned in the KSP chapter. The KSP was already written out in the chapter title for the KSP, so I kept that.

jans23 commented:
Thanks.

mmerklinger commented:
@jans23 I included the info about the module unload error with references. If everything is fine you can merge it now.

jans23 committed:
Clarify PKCS#11 access and execution context

Reworded the explanation of PKCS#11 access requirements for clarity and added that the PKI Proxy server and client can run on the same computer.
